src/fs.toit

```diff
@@ -3,75 +3,312 @@
+  example `c:`.
+
+Drive absolute paths are drive paths that are fully qualified (absolute).
+  For example `c:\\path\\to\\file`. They start with a drive letter and a
+  colon, followed by a path that is rooted on the drive.
+
+Drive relative paths are drive paths that are relative to the current
+  working directory of the drive. For example, `c:foo\\bar`. They start
+  with a drive letter and a colon, followed by a path that is *not*
+  rooted on the drive. This type of path is a relict from the DOS days
+  and should not be used anymore. On current versions of Windows there
+  isn't a drive-specific working directory anymore, although there are hidden
+  environment variables to indicate drive-specific current working
+  directories. If the drive of a drive-relative path is the same
+  as the current working directory, then the path is equivalent to a
+  relative path. If the drive is different, then the path is equivalent
+  to an absolute path (unless the environment variable is set).
+
+Rooted paths start with a path separator. These are rooted on the
+  drive of the current working directory. The full path is
+  obtained by prepending the root drive (or UNC path if set).
+  Example: `\\path\\to\\file`.
+
+Relative paths are paths that are relative to the current working
+  directory. They are not rooted
+
+Universal Naming Convention (UNC) absolute paths are, more or less, Window's
+  alternative to URLs. They are used to access remote
+  file systems (typically SMB). A network share
+  always starts with two separators. After the separators comes the
+  host (either a NetBIOS machine name, or an IP address), followed by
+  the share name. Together, this makes up the volume name of a UNC
+  path. Example: `\\\\host\\share\\path`.
+
+Local device paths are paths that start with the sequence "\\\\.\\"
+  (two backslashes, a dot, and another backslash). They are used to
+  access devices directly, bypassing the file system. For example
+  `\\\\.\\COM1` is the first serial port.
+
+## Volume name
+A volume name is the name of a drive, for example `C:` or `D:`, or a UNC
+  path, for example `\\\\host\\share`.
+
+## Examples
+Examples for Windows paths. Each example is followed by the result of calling $is-absolute, $is-rooted, and $is-relative on the path.
+- `C:\\Documents\\Newsletters\\Summer2018.pdf` - an absolute file path from the root of drive `C:`.
+- `\\Program Files\\Custom Utilities\\StringFinder.exe` - a relative path from the root of the current drive.
+- `2018\\January.xlsx` - a relative path to a file in a subdirectory of the current directory.
+- `..\\Publications\\TravelBrochure.pdf` - a relative path to a file in a directory starting from the current directory.
+- `C:\\Projects\\apilibrary\\apilibrary.sln` - an absolute path to a file from the root of drive `C:`.
+- `C:Projects\\apilibrary\\apilibrary.sln` - a relative path from the current directory oft he `C:` drive.
+*/
+
+is-windows_ -> bool:
```

Comment:
Would it be terrible if this was public? I wouldn't mind seeing `fs.is-windows` in some code even though it probably isn't ideal to use it in almost all cases.

I'd probably cache the result of the comparison if so.

Reply:
We could add it to `system`, but I don't think the `fs` library should do this job.

Stored in static.
done.